WIP BXMSDOC-8000 & BXMSDOC-8001: Update Kogito book as per 7.12 STP BAPLs (#72)
hmanwani-rh/kie-docs: doc-content/enterprise-only/getting-started-kogito/con-kogito-process-decisions-integration.adoc

[id="con-kogito-process-decisions-integration_{context}"]
= Integration of process and decision microservices in {KOGITO}

[role="_abstract"]

> When designing a business process in {KOGITO}, you can define the decisions as distinct microservices using DMN and DRL. In {KOGITO}, you can integrate the processes and decisions using the following methods:

Reviewer: ```suggestion
When designing a business process in {KOGITO}, you can use DMN and DRL to define the decisions as distinct microservices. In {KOGITO}, you can use the following methods to integrate the processes and decisions:
```

Author: Done

> * *Embedded method*: Allows the integration of process and decision microservices using business rule tasks.

Reviewer: ```suggestion
* *Embedded method*: Enables you to integrate process and decision microservices using business rule tasks.
```

Author: Done

> * *Remote method*: Allows the integration of process and decision microservices using REST endpoints.

Reviewer: ```suggestion
* *Remote method*:  Enables you to integrate process and decision microservices using REST endpoints.
```

Author: Done

> You can either use the embedded method or remote method based on the use cases, or you can combine both methods in a single process.

Reviewer: ```suggestion
Depending on the use case, you can use either the embedded method or the remote method, or you can combine both methods in a single process.
```

Author: Done

[id="con-kogito-process-decisions-integration_{context}"]
= Integration of process and decision microservices in {KOGITO}

[role="_abstract"]
When designing a business process in {KOGITO}, you can define the decisions as distinct microservices using DMN and DRL. In {KOGITO}, you can integrate the processes and decisions using the following methods:

* Embedded method
* Remote method

You can either use the embedded method or remote method based on the use cases, or you can combine both methods in a single process.

> == Embedded method to integrate process and decision microservices in {KOGITO}

Reviewer: ```suggestion
==  The embedded method of integrating process and decision microservices in {KOGITO}
```

Author: How about if we avoid the title starting with an article:

Embedded method of integrating process and decision microservices in {KOGITO}

> The embedded method to integrate the processes and decisions uses business rule tasks. The business rule tasks access the communication between a local process and {DECISION_ENGINE} that is running in the same application. The embedded method requires the decision microservices to be deployed in the same application, in which the process microservice is running.

Reviewer: ```suggestion
The embedded method of integrating processes and decisions uses business rule tasks. Business rule tasks access communication between a local process and a  {DECISION_ENGINE} that is running in the same application. The embedded method requires the decision microservices to be deployed in the same application in which the process microservice is running.
```

Author: Done

> The embedded method to integrate the processes and decisions uses business rule tasks. The business rule tasks access the communication between a local process and {DECISION_ENGINE} that is running in the same application. The embedded method requires the decision microservices to be deployed in the same application, in which the process microservice is running.

Reviewer: I would prefer that you not use the passive voice above, but I think you are avoiding making this a procedure? So your call.

Author: Yes, I want this section to be a conceptual module since this section describes more of the concepts of the embedded method.

For more information about business rule tasks, see {URL_DEVELOPING_PROCESS_SERVICES}#bpmn-tasks-overview-con_business-processes[_BPMN2 tasks in process designer_].

> .Embedded method to integrate a process with decisions

Reviewer: ```suggestion
.The embedded method of integrating a process with decisions
```

image::KogitoMicroservices/process-decisions-embedded-integration.png[Image of Embedded process integration with decisions]

> You can also use `process-decisions-quarkus` and `process-decisions-springboot` example applications for {QUARKUS} and Spring Boot.

Reviewer: ```suggestion
You can also use  the `process-decisions-quarkus` and `process-decisions-springboot` example applications for {QUARKUS} and Spring Boot, retrospectively.

Author: I think it would be "respectively" instead of retrospectively.

> In the further sections, you can see an example of the traffic violations project, which describes decision microservices within a process microservice and how to perform integration using business rule tasks. in {KOGITO}, the decision microservices are defined in different domains or assets, such as DMN and DRL. However, all the assets including BPMN, DMN, and DRL must be stored in the `/resources` folder.

Reviewer: ```suggestion
 In the following section, you can see an example of the traffic violations project, which describes decision microservices within a process microservice and how to perform integration using business rule tasks.I n {KOGITO}, the decision microservices are defined in different domains or assets, such as DMN and DRL. However, all the assets including BPMN, DMN, and DRL must be stored in the `/resources` folder.
```

Author: Done

> === Example traffic violations project for embedded method

Reviewer: ```suggestion
=== Example traffic violations project for the embedded method
```

Author: Done

> The traffic violations project is based on the evaluation process of the traffic violations. In this project, first the information of the driver is fetched, and based on the fetched information, the driving license of the driver is validated using a RuleUnit in DRL. After the driving license validation, the violation evaluation is executed that is defined as a DMN decision microservice. The output contains the information whether the driver is suspended or not.

Reviewer: ```suggestion
The traffic violations project is based on the evaluation process of traffic violations. In this project, first the information of the driver is fetched, and based on the fetched information, the driver's license of the driver is validated using a RuleUnit in DRL. After the driver's license is validated, the violation evaluation is executed that is defined as a DMN decision microservice. The output contains information about whether or not the driver is suspended.
```

Author: Done
